conda/gateways/repodata/__init__.py

```diff
@@ -289,3 +290,88 @@ def conda_http_errors(url, repodata_fn):
             e.response,
             caused_by=e,
         )
+
+
+class CacheJsonState(dict):
+    """
+    Load/save `.state.json` that accompanies cached `repodata.json`
+    """
+
+    def __init__(self, cache_path_json, cache_path_state, repodata_fn):
+        self.cache_path_json = pathlib.Path(cache_path_json)
+        self.cache_path_state = pathlib.Path(cache_path_state)
+        self.repodata_fn = repodata_fn
+
+    def load(self):
+        """
+        Cache headers and additional data needed to keep track of the cache are
+        stored separately, instead of the previous "added to repodata.json"
+        arrangement.
+        """
+        try:
+            state_path = self.cache_path_state
+            log.debug("Load %s cache from %s", self.repodata_fn, state_path)
+            with state_path.open("r") as s:
+                state = json.load(s)
```

Comment:
```suggestion
            log.debug("Load %s cache from %s", self.repodata_fn, self.cache_path_state)
            state = json.loads(self.cache_path_state.read_text())
```

Reply:
This is probably wrong, since I should be using the new class?

Reply:
this is the new state class?